### PR TITLE
[BACKLOG-312] - Use OSGi as a modular container for data lineage project

### PR DESCRIPTION
**Pull request by rfellows**

### CODE REVIEW

**Comment by rfellows** [comment]
@pentaho-nbaker please review. it depends on this one https://github.com/pentaho/pentaho-metaverse-ee/pull/225 being merged/built so the bundle will be published.

File: pentaho-karaf-features/src/main/resources/pentaho-karaf-features-enterprise.xml [hunk 1]
```diff
@@ -115,4 +115,46 @@
...
 
+  <feature name="pentaho-commons-configuration" version="1.6">
+    <bundle>mvn:commons-configuration/commons-configuration/1.6</bundle>
+    <bundle>mvn:commons-jxpath/commons-jxpath/1.3</bundle>
+    <bundle>wrap:mvn:commons-digester/commons-digester/1.8</bundle>
+    <bundle>mvn:commons-beanutils/commons-beanutils/1.8.3</bundle>
+    <bundle>mvn:commons-codec/commons-codec/1.6</bundle>
+    <bundle>mvn:commons-collections/commons-collections/3.2.1</bundle>
+    <bundle>wrap:mvn:jdom/jdom/1.0</bundle>
+    <bundle>wrap:mvn:org.apache.ant/ant/1.7.0</bundle>
```
Comment (pentaho-nbaker) [inline pentaho-karaf-features/src/main/resources/pentaho-karaf-features-enterprise.xml]: What's up with Ant?

Comment (rfellows) [inline pentaho-karaf-features/src/main/resources/pentaho-karaf-features-enterprise.xml]: without it, we get this
Error executing command: Error starting bundles:
	Unable to start bundle 245: Unresolved constraint in bundle pentaho-metaverse-ee [245]: Unable to resolve 245.0: missing requirement [245.0] osgi.wiring.package; (&(osgi.wiring.package=org.apache.commons.configuration)(version>=1.6.0)(!(version>=2.0.0))) [caused by: Unable to resolve 155.0: missing requirement [155.0] osgi.wiring.package; (osgi.wiring.package=org.apache.tools.ant.taskdefs)]

**Comment by rfellows** [comment]
@pentaho-nbaker updated to a new version of commons config. now we don't need a feature definition at all for it.

**Pull request merged**